settings.gradle.kts

```diff
@@ -8,13 +8,15 @@ pluginManagement {
 }
 
 dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
     repositories {
         mavenCentral()
     }
 }
 
 plugins {
     `gradle-enterprise`
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.6.0"
```

Comment:
Shouldn't this be managed via the libs.versions.toml? And is it correct that the plugin is required in both the `root` and in `build-logic` version of `settings.gradle.kt`.

Reply:
>And is it correct that the plugin is required in both the root and in build-logic version of settings.gradle.kt.

Unfortunately yes. I can describe a scenario to demonstrate when each foojay plugin will be used.
Assuming a person has only Java 18 installed locally and wants to run `testJdk11` task, gradle will:
1. Read `/`'s settings.gradle, and will conclude it has to compile `includedBuild` first:
2. Gradle moves its focus to `/build-logic` and attempts to compile it first 
3. it will realize it is missing Java 20, (`/build-logic` sets the requirement via `jvmToolchain(20)` in `/build-logic/build.gradle.kts` <- this is where build-logic's foojay plugin steps in and fetches and configures required Java
2. As a result of successful compilation, it produces `build-logic.jar` with bytecode compatible with corresponding `jvmTarget`
![image](https://github.com/pinterest/ktlint/assets/36954793/8899dea1-89b5-4d2a-965c-cb6f817b6504)

4. Gradle will then move back t o`/`. It will eventual learn that `/` also requires Java 20 (via the other `jvmToolchain` call, but it has it installed already, done. 
5. Next important step is it's time to configure the requested task, it has `javaLauncher` set to use Java 11, Gradle _sees_ it has only Java 18 and Java 20 available, so it has to fetch the missing one.  <- this is where the `/`s foojay plugin steps in

If I skipped one of the plugin application it could result in a configuration error, and the only workaround it would be  to require to installing Java X manually. Having both plugins installed, guarantees in all cases the whole required setup configures itself automatically